### DISCUSSION TITLE
How can I get the type of the reponse object?

### DISCUSSION THREAD

**Discussion by johhnry** [socketio/socket.io#4105] [Q&A] [answered]
Hi,

I am using socket.io-client with TypeScript and I am using acknowledgements when emitting a message:

```javascript
socket.emit("update item", "1", { name: "updated" }, (response) => {
  console.log(response.status);
});
```
(from the [doc](https://socket.io/docs/v4/emitting-events/#acknowledgements))

But I have the following error in TS:

```
Parameter 'response' implicitly has an 'any' type.  TS7006

    17 |   useEffect(() => {
    18 |     socket.on("connect", () => {
  > 19 |       socket.emit("update item", "1", { name: "updated" }, (response) => {
       |                                                             ^
    20 |         console.log(response.status);
    21 |       });
    22 |     });
```

How can I type the response object? Where can I find it? The type of `emit` has `..args: EventParams<EmitEvents, Ev>`

I want to be able to get the status or the error field.

Thanks!

**Comment by darrachequesne** [comment, marked as answer]
Hi! Please check this: https://socket.io/docs/v4/migrating-from-3-x-to-4-0/#typed-events

I'll update the documentation with a more complete example.

**Reply by johhnry** [reply]
Thank you so much I didn't see it!

**Reply by johhnry** [reply]
@darrachequesne 
Also when I want to unsuscribe a listener with the `off` method, the first parameter is an optional `string` but it should also be typed isn't it?

Currently it's:

```typescript
interface Emitter<Event = string> {
    off(event?: Event, listener?: Function): Emitter;
}
```

But it should use the same as `on`?:

```typescript
abstract class StrictEventEmitter<...> extends Emitter {
  on<Ev extends ReservedOrUserEventNames<ReservedEvents, ListenEvents>> //...
}
```
